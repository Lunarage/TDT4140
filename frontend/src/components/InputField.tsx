import React from "react";
import { Input } from "semantic-ui-react";
import styled from 'styled-components';
import { redHexColor } from "../consts";

const LineWrapper = styled.div`
  margin: 8px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 25px;
`;

const BoldText = styled.div`
  font-size: 15px;
  font-weight: bold;
  color: ${redHexColor};
`;
interface InputFieldProps {
  name: string;
  password?: boolean;
  text?: string;
  onChangeFunc: (val: string) => void
}

const InputField = ({ name, password, text, onChangeFunc }: InputFieldProps) => {
  return (
    <LineWrapper>
      <BoldText>{name}</BoldText>
      <Input
        size='massive'
        placeholder={text}
        onChange={(event) => onChangeFunc(event.target.value)}
      />
    </LineWrapper>
  )
}
export default InputField;
