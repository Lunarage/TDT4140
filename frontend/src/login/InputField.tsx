import { Input } from "semantic-ui-react";
import styled from 'styled-components';
import { redHexColor } from "../consts";

const LineWrapper = styled.div`
  margin: 15px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 25px;
`;

const BoldText = styled.div`
  font-size: 15px;
  font-weight: bold;
  color: ${redHexColor};
`;
interface InputFieldProps {
  name: string;
  type?: string;
  text?: string;
  onChangeFunc: (val: string) => void
}

// Used for login / register input fields
const InputField = ({ name, type, text, onChangeFunc }: InputFieldProps) => {
  return (
    <LineWrapper>
      <BoldText>{name}</BoldText>
      <Input
        type={type}
        placeholder={text}
        onChange={(event) => onChangeFunc(event.target.value)}
      />
    </LineWrapper>
  )
}
export default InputField;
