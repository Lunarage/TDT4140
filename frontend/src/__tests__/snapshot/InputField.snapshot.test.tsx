import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import InputField from '../../login/InputField';

Enzyme.configure({ adapter: new Adapter() });

describe("InputField", () => {
  it("renders InputField", () => {
    shallow(<InputField name="test" onChangeFunc={() => { }} />);
  });
  it("Should match InputField snapshot", () => {
    const snap = shallow(<InputField name="test" onChangeFunc={() => { }} />);
    expect(toJSON(snap)).toMatchSnapshot();
  });
});
