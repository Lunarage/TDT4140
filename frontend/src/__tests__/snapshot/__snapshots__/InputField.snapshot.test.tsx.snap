// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Should match InputField snapshot 1`] = `
<styled.div>
  <styled.div>
    test
  </styled.div>
  <Input
    onChange={[Function]}
    type="text"
  />
</styled.div>
`;
